client/jetpack-cloud/sections/agency-signup/agency-signup-form/index.tsx

```diff
@@ -0,0 +1,99 @@
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import CardHeading from 'calypso/components/card-heading';
+import CompanyDetailsForm from 'calypso/jetpack-cloud/sections/partner-portal/company-details-form';
+import { formatApiPartner } from 'calypso/jetpack-cloud/sections/partner-portal/utils';
+import { partnerPortalBasePath } from 'calypso/lib/jetpack/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { errorNotice, removeNotice } from 'calypso/state/notices/actions';
+import { receivePartner } from 'calypso/state/partner-portal/partner/actions';
+import useCreatePartnerMutation from 'calypso/state/partner-portal/partner/hooks/use-create-partner-mutation';
+import { getCurrentPartner } from 'calypso/state/partner-portal/partner/selectors';
+import { translateInvalidPartnerParameterError } from 'calypso/state/partner-portal/partner/utils';
+import type { APIError } from 'calypso/state/partner-portal/types';
+import type { ReactElement } from 'react';
+import './style.scss';
+
+export default function AgencySignupForm(): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const partner = useSelector( getCurrentPartner );
+	const notificationId = 'partner-portal-agency-signup-form';
+
+	const createPartner = useCreatePartnerMutation( {
+		onSuccess: ( partner ) => {
+			dispatch( receivePartner( formatApiPartner( partner ) ) );
+
+			page.redirect( partnerPortalBasePath() );
+		},
+		onError: ( error: APIError ) => {
+			let message = error.message;
+
+			if ( error.code === 'rest_invalid_param' && typeof error?.data?.params !== undefined ) {
```

Comment:
Did you mean `'undefined'` rather than `undefined`?

Reply:
Nice catch on these! Tracking in 1202074155672006-as-1202436394788720